CMakeLists.txt

```diff
@@ -182,7 +182,7 @@ configure_file(
   ${CMAKE_CURRENT_BINARY_DIR}/h5vl_test_config.h
 )
 
-add_subdirectory(driver)
```

Comment:
We have to be careful about permanently commenting out this functionality because it's still used by the DAOS VOL connector for example. Over in https://github.com/HDFGroup/hdf5/blob/feature/api_tests/test/API/CMakeLists.txt#L44-L46 I've surrounded it with a CMake option, but the DAOS VOL would still need to be updated for that as well. We should figure out why you're not able to compile the driver program correctly.

Reply:
I'll look into getting the driver to compile and re-add this.

Reply:
Solution was to declare `CFLAGS = -D_POSIX_C_SOURCE=199506L` as an environment variable before invoking cmake. Just using it for the following `make` command wasn't sufficient, and `-D_POSIX_C_SOURCE=1` still left some symbols undefined.